library/item/item_setting/src/main/java/org/quiltmc/qsl/itemsetting/api/EquipmentSlotProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.itemsetting.api;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.item.ItemStack;
+
+/**
+ * A provider for the preferred {@link EquipmentSlot} of an item.
```

Comment:
What does  "preferred" mean in this context? Does it mean that an EquipmentSlotProvider can be put in any slot, but just prefers this one?